packages/extension-utils/src/OAuthConfigForm/src/types.ts

```diff
@@ -0,0 +1,41 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2023 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import type { ValidationMessages, MessageBarIntent } from '@looker/components'
+import type { ConfigValues } from './utils'
+
+export interface OAuthFormState {
+  apiServerUrlValue: string
+  fetchedUrl: string
+  webUrlValue: string
+  messageBar: MessageBarData
+  validationMessages: ValidationMessages
+  savedConfig: ConfigValues
+}
+
+export interface MessageBarData {
+  intent: MessageBarIntent
+  text: string
+}
```

Comment:
I'd prefer it if types didn't live in a separate types.ts file. It helps keep things from getting bloated.

Reply:
I just want to avoid declaring it in two places. Because the OAuthFormState is imported in the test file as well. Would you just export it from the OAuthForm?

Reply:
Yes, in that case export it from its primary use area